### ISSUE TITLE
js syntax.Incomplete sometimes incorrect

### ISSUE THREAD

**Issue by rob-myers** [mvdan/sh#308]
Given code represented as a string, I want to know whether it is incomplete or not.
Your new API works in some cases (first 3 below), but not others (last 3 below).

Each case works in `gosh`, although `echo \` shows the wrong prompt.

```js
const { syntax } = require("mvdan-sh");

function isIncomplete(code) {
  let parser = syntax.NewParser(), incomplete;
  try {
    parser.Interactive(
      { read: () => code },
      (stmts) => (incomplete = parser.Incomplete()) && false,
    );
  } catch (e) {}
  console.log(`${code} ${incomplete ? 'incomplete' : 'complete'}`);
}

// incomplete (correct)
isIncomplete('echo "foo\n');
// incomplete (correct, string has same value)
isIncomplete('echo \"foo\n');
// complete (correct, quote is escaped)
isIncomplete('echo \\"foo\n');
// complete (incorrect, but gosh correct)
isIncomplete('echo foo |\n');
// complete (incorrect, but gosh correct)
isIncomplete('echo foo &&\n');
// complete (incorrect, gosh shows wrong prompt)
isIncomplete('echo foo \\\n');
```

p.s. Reposting this issue, since previous disappeared.

**Comment by mvdan** [comment]
Please note that GitHub has been badly broken for over twelve hours, so it's been hard for me to even read these issues.

It looks like your issues are back up, so I'm closing this one in favor of the older one.

**Comment by rob-myers** [comment]
Indeed, I eventually noticed. However I still can't see the other one...

**Comment by mvdan** [comment]
Try waiting another hour. The other issue is closed too, as I've fixed all of the bugs there and pushed v0.3.0 to NPM.
